swift/Samples/Tutorial/Tutorial3.md

```diff
@@ -0,0 +1,645 @@
+# Step 3
+
+_State throughout a tree of workflows_
+
+## Setup
+
+Start from the completed code from Tutorial 2, or if you're jumping ahead, the completed `Tutorial2` pod.
+
+## Editing TODO items
+
+Now that a user can "log in" to their TODO list, we want to add the ability to edit the TODO items listed.
+
+### State ownership
+
+In the workflow framework, state flows _down_ the tree, and actions come _up_ as output events (as in the traditional computer science trees that grow downward).
```

Comment:
Perhaps:

> flows _down_ the tree (in the form of the properties provided by the parent workflow)

Reply:
"actions come up" is weird, since action really means workflow state delta function.

Reply:
Updating to say "data flows _down_ the tree as properties to child workflows, and actions come _up_ as output events". The spirit of this is to describe it in the same was as React, which has the common description of "data down, actions up"